### PR TITLE
Integrate flow-based model with conditioning on LinearPhysics class of inverse problem 

### PR DESCRIPTION
**Pull request by MaximeSongIdris**
### Checks to be done before submitting your PR
- [ ] `python3 -m pytest tests/` runs successfully.
- [ ] `black .` runs successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [ ] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/loss/rf_loss.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    TODO: complete this and missing example of usage
+    """
+
+    def __init__(self, model):
+        super().__init__()
+        self.nn_model = model
+
+    def forward(self, y, physics, x_gt, **kwargs):
+        r"""
+        Generate the forward process at timestep t for the Rectified Flow model and get output of the model.
```
Comment (tachella) [inline deepinv/loss/rf_loss.py:31]:
```suggestion
        Run the flow matching model forward process.
```

File: deepinv/loss/rf_loss.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        # For reconstruction network
+        x_net = self.nn_model(y=z_t, physics=t_diffusion_physics)
+
+        return x_net
+
+    @classmethod
+    @torch.no_grad()
+    def sample(cls, model, y, physics, sample_steps=50, cfg=2.0):
+        r"""
+        ODE solver for the RF model.
```
Comment (tachella) [inline deepinv/loss/rf_loss.py:61]: change to
```suggestion
        ODE solver for the flow matching model.
```

File: deepinv/loss/rf_loss.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            # estimation of the velocity from y to x
+            vc = (x_hat - y) / t
+
+            y = y + dt * vc
+            images.append(y)
+
+        return images
+
+
+class RFLoss(Loss):
```
Comment (tachella) [inline deepinv/loss/rf_loss.py:100]:
```suggestion
class FlowMatchingLoss(Loss):
```

File: deepinv/loss/rf_loss.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    r"""
+    `Standard Rectified Flow loss <https://github.com/cloneofsimo/minRF>`_
```
Comment (tachella) [inline deepinv/loss/rf_loss.py:102]:
```suggestion
    `Standard flow matching loss <https://github.com/cloneofsimo/minRF>`_
```

File: deepinv/loss/rf_loss.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    TODO !!!
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.name = "Rectified Flow loss"
```
Comment (tachella) [inline deepinv/loss/rf_loss.py:109]:
```suggestion
        self.name = "Flow matching loss"
```

File: deepinv/physics/forward.py [hunk 6]
```diff
@@ -385,86 +427,185 @@ def A_adjoint_A(self, x, **kwargs):
...
+            tol=self.tol,
         )
-        noise = self.noise_model
-        sensor = self.sensor_model
+
+    def __mul__(self, other):
+        r"""
+        Compose two linear forward operators :math:`A = A_1\circ A_2` via the * operation.
+
+        0) We have `new_sensor_model` = `self.sensor_model`.
```
Comment (tachella) [inline deepinv/physics/forward.py:483]:
```suggestion
        0) We have ``new_sensor_model`` = ``self.sensor_model``.
```

**Review by tachella — CHANGES_REQUESTED** [review]
The PR is looking good, some remaining fixes:

- merge from main to have the new stacked physics, and remove the old one.
- be careful with using two accents to have `code style` instead of _italics_ in restructured text
- rename rectified flow to flow matching
- add mul operation to the general Physics class (for the case of non-linear physics)
- update to new docs, add the loss to `docs/user_guide/training/loss.rst` and `docs/api/loss.rst`

**Comment by matthieutrs** [comment]
Closing this PR for now as we did not resolve use cases with FM loss, we will reopen in the future. In the mean time, fixes of relevance that need to merged have been moved to #384
